lib/python/picongpu/picmi/simulation.py

```diff
@@ -28,10 +31,85 @@ class Simulation(picmistandard.PICMI_Simulation):
+
+    picongpu_moving_window_move_point = pypicongpu.util.build_typesafe_property(typing.Optional[float])
+    """
+    point a light ray reaches in y from the left border until we begin sliding the simulation window with the speed of
+    light
+
+    in multiples of the simulation window size
+
+    @attention if moving window is active, one gpu in y direction is reserved for initializing new spaces,
+        thereby reducing the simulation window size accordingrelative spot at which to start moving the simulation window
+    """
+
+    picongpu_moving_window_stop_iteration = pypicongpu.util.build_typesafe_property(typing.Optional[int])
+    """iteration, at which to stop moving the simulation window"""
+
+    __runner = pypicongpu.util.build_typesafe_property(typing.Optional[pypicongpu.runner.Runner])
+
+    # @todo remove boiler plate constructor argument list once picmistandard reference implementation switches to
+    #   pydantic, Brian Marre, 2024
+    def __init__(
+        self,
+        picongpu_template_dir: typing.Optional[typing.Union[str, pathlib.Path]] = None,
+        picongpu_typical_ppc: typing.Optional[int] = None,
+        picongpu_moving_window_move_point: typing.Optional[float] = None,
+        picongpu_moving_window_stop_iteration: typing.Optional[int] = None,
+        picongpu_interaction: typing.Optional[Interaction] = None,
+        **keyword_arguments,
+    ):
+        if picongpu_template_dir is not None:
+            self.picongpu_template_dir = str(picongpu_template_dir)
+        else:
+            self.picongpu_template_dir = picongpu_template_dir
+
+        self.picongpu_typical_ppc = picongpu_typical_ppc
+        self.picongpu_moving_window_move_point = picongpu_moving_window_move_point
+        self.picongpu_moving_window_stop_iteration = picongpu_moving_window_stop_iteration
+        self.picongpu_interaction = picongpu_interaction
+        self.picongpu_custom_user_input = None
+        self.__runner = None
+
+        # second call PICMI __init__ to do PICMI initialization and setting class attribute values outside of pydantic model
+        picmistandard.PICMI_Simulation.__init__(self, **keyword_arguments)
+
+        # additional PICMI stuff checks, @todo move to picmistandard, Brian Marre, 2024
+        ## throw if both cfl & delta_t are set
+        if self.solver is not None and "Yee" == self.solver.method and isinstance(self.solver.grid, Cartesian3DGrid):
+            self.__yee_compute_cfl_or_delta_t()
+
+        # checks on picongpu specific stuff
+        ## template_path is valid
+        if picongpu_template_dir == "":
+            raise ValueError("picongpu_template_dir MUST NOT be empty string")
+        if picongpu_template_dir is not None:
+            template_path = pathlib.Path(picongpu_template_dir)
+            if not template_path.is_dir():
+                raise ValueError("picongpu_template_dir must be existing directory")
```

Comment:
because the setter is auto generated by the `build_type_safe_property` and should not be overwritten, it also follows the current code style of the picongpu PICMI interface and PICMI standard in general.

I admittedly don't particularly like the style and would prefer to change it, but following the expectation was probably better in this case.